File: src/main/java/io/puharesource/mc/titlemanager/api/v2/animation/AnimationPart.java
package io.puharesource.mc.titlemanager.api.v2.animation;

/**
 * A class used to create animations from strings and animations.
 *
 * @param <T> Should be a {@link String} or {@link Animation}.
 *
 * @since 2.0.0
 */
public interface AnimationPart<T> {
    /**
     * @return The part that should be displayed.
     */
    T getPart();
}

File: src/main/kotlin/io/puharesource/mc/titlemanager/api/v2/animation/AnimationPart.kt
package io.puharesource.mc.titlemanager.api.v2.animation

/**
 * A class used to create animations from strings and animations.
 *
 * @param <T> Should be a [String] or [Animation].
 *
 * @since 2.0.0
</T> */
fun interface AnimationPart<T> {
    /**
     * The part that should be displayed.
     */
     fun getPart(): T
}
